### PR TITLE
Add reload functinality

### PR DESCRIPTION
**Pull request by weeryan17**
The title says it all

### CODE REVIEW

File: src/main/java/io/github/binaryoverload/JSONConfig.java [hunk 1]
```diff
@@ -52,6 +52,8 @@
  */
 public class JSONConfig {
 
+    private int mode; // 0 = File, 1 = String, 2 = Input Stream, -1 = JsonObject
```
Comment (WalshyDev) [inline src/main/java/io/github/binaryoverload/JSONConfig.java]: why even have this? You could just set a Reader for all of these.

Comment (weeryan17) [reply src/main/java/io/github/binaryoverload/JSONConfig.java]: The mode determines the read i use.

Comment (WalshyDev) [reply src/main/java/io/github/binaryoverload/JSONConfig.java]: Yeah why bother, you can use a Reader for all of these. No need for a weird mode thing and an Object file

File: src/main/java/io/github/binaryoverload/JSONConfig.java [hunk 2]
```diff
@@ -52,6 +53,7 @@
  */
 public class JSONConfig {
 
+    Reader reader;
```
Comment (WalshyDev) [inline src/main/java/io/github/binaryoverload/JSONConfig.java]: private

Comment (weeryan17) [reply src/main/java/io/github/binaryoverload/JSONConfig.java]: 👀

File: src/main/java/io/github/binaryoverload/JSONConfig.java [hunk 3]
```diff
@@ -74,7 +76,8 @@
      */
     public JSONConfig(File file) throws FileNotFoundException, NullPointerException {
         Objects.requireNonNull(file);
-        this.object = GSON.fromJson(new JsonReader(new FileReader(file)), JsonObject.class);
+        reader = new FileReader(file);
+        this.object = GSON.fromJson(new JsonReader(reader), JsonObject.class);
```
Comment (binaryoverload) [inline src/main/java/io/github/binaryoverload/JSONConfig.java]: Either do `this.reader` as well or have it as `object`. I recommend the second

**Pull request merged**